modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/query/SolrLuceneQueryConverter.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.search.solr8.internal.query;
+
+import com.liferay.portal.kernel.search.Query;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Joao Victor Alves
+ */
+@Component(
+	property = "search.engine.impl=Solr", service = LuceneQueryConverter.class
+)
+public class SolrLuceneQueryConverter
+	extends SolrQueryTranslator implements LuceneQueryConverter {
+
```

Comment:
Hey @MrJohn1911, you do not need to extend the  `SolrQueryTranslator ` here. Try to just implement the `QueryVisitor<org.apache.lucene.search.Query>` instead. Thx

Reply:
Hi @rafaprax, instead of doing that, I created a BaseQueryVisitor and extended it in both classes, because both of them needed to be a QueryVisitor so the methods `convert` and `translate` could work.